sys/cddl/contrib/opensolaris/uts/common/dtrace/dtrace.c

```diff
@@ -13778,29 +13779,52 @@ dtrace_dof_predicate(dof_hdr_t *dof, dof_sec_t *sec, dtrace_vstate_t *vstate,
 	return (dtrace_predicate_create(dp));
 }
 
+static dof_actdesc_t
+freebsd64_dof_actdesc(dof_actdesc64_t *actdesc64)
+{
+	dof_actdesc_t actdesc;
+	actdesc.dofa_difo = actdesc64->dofa_difo;
+	actdesc.dofa_strtab = actdesc64->dofa_strtab;
+	actdesc.dofa_kind = actdesc64->dofa_kind;
+	actdesc.dofa_ntuple = actdesc64->dofa_ntuple;
+	actdesc.dofa_arg = actdesc64->dofa_arg;
+	actdesc.dofa_uarg = (uintcap_t)cheri_fromint(actdesc64->dofa_uarg);
+	return actdesc;
+}
+
 static dtrace_actdesc_t *
 dtrace_dof_actdesc(dof_hdr_t *dof, dof_sec_t *sec, dtrace_vstate_t *vstate,
     cred_t *cr)
 {
 	dtrace_actdesc_t *act, *first = NULL, *last = NULL, *next;
-	dof_actdesc_t *desc;
+	dof_actdesc_t _desc;
+	dof_actdesc_t *desc = &_desc;
```

Comment:
Why not just make `desc` the actual object, rather than this (and even if your approach were to be used, `_desc` is not a good name).

Reply:
was to avoid to change all the accesses to disc. But there aren't many..
If this was the correct approach, what would have been a good name then?

Reply:
Well, yes, there are many accesses, I think this is the best way not to change `->` in `.` in a lot of lines

Reply:
?